crates/router/src/core/recon.rs

```diff
@@ -0,0 +1,291 @@
+) -> RouterResponse<recon_api::ReconStatusResponse> {
+    let global_db = &*state.global_store;
+    let db = &*state.store;
+    let key_manager_state = &(&state).into();
+
+    let user_role = db
+        .find_user_role_by_user_id_and_lineage(
+            &user.user_id,
+            &user.org_id,
+            &user.merchant_id,
+            user.profile_id.as_ref(),
+            UserRoleVersion::V1,
+        )
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
+    let merchant_id = user_role
+        .merchant_id
+        .ok_or(errors::ApiErrorResponse::InternalServerError)?;
+    let key_store = db
+        .get_merchant_key_store_by_merchant_id(
+            key_manager_state,
+            &merchant_id,
+            &db.get_master_key().to_vec().into(),
+        )
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
+
+    // Ensure the merchant in user role is the parent of the business profile
+    match user.profile_id {
+        Some(profile_id) => {
+            let business_profile = db
+                .find_business_profile_by_profile_id(key_manager_state, &key_store, &profile_id)
+                .await
+                .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
+
+            if user.merchant_id != merchant_id || user.merchant_id != business_profile.merchant_id {
+                Err(report!(errors::ApiErrorResponse::PreconditionFailed {
+                    message: "Profile id authentication failed. Please use the correct JWT token"
+                        .to_string(),
+                }))
+            } else {
+                Ok(())
+            }
+        }
+        None => Ok(()),
+    }?;
+
+    let merchant_account = db
+        .find_merchant_account_by_merchant_id(key_manager_state, &merchant_id, &key_store)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
+
+    let user_from_db = global_db
+        .find_user_by_id(&user.user_id)
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)?;
+    let email_contents = email_types::ProFeatureRequest {
+        feature_name: "RECONCILIATION & SETTLEMENT".to_string(),
+        merchant_id: merchant_id.clone(),
+        user_name: UserName::new(user_from_db.name)
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Failed to form username")?,
+        recipient_email: UserEmail::new(Secret::new("biz@hyperswitch.io".to_string()))
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Failed to convert recipient's email to UserEmail")?,
+        settings: state.conf.clone(),
+        subject: format!(
+            "Dashboard Pro Feature Request by {}",
+            user_from_db.email.expose().peek()
+        ),
+    };
+
+    let is_email_sent = state
```

Comment:
if youre not updating `merchant_account` table if email is not sent, then why can we just propagate this error ?